server/src/main/java/org/elasticsearch/cluster/BatchedSimpleTaskExecutor.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.core.Tuple;
+
+/**
+ * A {@link BatchedTaskExecutor} for tasks that do not return any value as a result of their execution, other than the next cluster state.
+ */
+public abstract class BatchedSimpleTaskExecutor<Task extends ClusterStateTaskListener> extends BatchedTaskExecutor<Task, Void> {
```

Comment:
I wonder if we really need to hide the `TaskResult` away like this. Could we just explicitly use `Void` and `null` in the places where there's no result? We use `ActionListener<Void>` similarly for example. I'd prefer to be explicit so that it's easier for folk to see that a result _can_ be passed around if needed.

Reply:
Sure, we can. I just found using a `Tuple` with `null` and the `Void` a bit useless. The `Void` part is ok as you mentioned. The `Tuple` part was a bit ugly. But I can change it if you find that not that bad. (I do agree with the less hiding argument).

Reply:
Yeah it's a little ugly but I think we can live with it. If it turns out to be something folks struggle with then we can revisit this.

Reply:
Done in https://github.com/elastic/elasticsearch/pull/90459/commits/1d527f5c6fd219a9ff5460d1d3805c5710b7cb6b